File: app/src/main/java/br/com/levimendesestudos/spidermagazine/utils/InternetUtil.java
package br.com.levimendesestudos.spidermagazine.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class InternetUtil {

    public static boolean isConnectedToInternet(Context context) {
        ConnectivityManager conMgr = (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo netInfo = conMgr.getActiveNetworkInfo();

        return netInfo != null && netInfo.isConnectedOrConnecting();
    }
}

File: app/src/main/java/br/com/levimendesestudos/spidermagazine/utils/InternetUtil.kt
package br.com.levimendesestudos.spidermagazine.utils

import android.content.Context
import android.net.ConnectivityManager

object InternetUtil {

    fun isConnectedToInternet(context: Context): Boolean {
        val conMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val netInfo = conMgr.activeNetworkInfo

        return netInfo != null && netInfo.isConnectedOrConnecting
    }
}